src/hera/workflows/_mixins.py

```diff
@@ -616,23 +616,24 @@ class CallableTemplateMixin(ArgumentsMixin):
     not duplicated.
     """
 
+    directly_callable: bool = False
+
     def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
         if "name" not in kwargs:
             kwargs["name"] = self.name  # type: ignore
 
-        arguments = self._get_arguments(**kwargs)
-        parameter_names = self._get_parameter_names(arguments)
-        artifact_names = self._get_artifact_names(arguments)
-
         # when the `source` is set via an `@script` decorator, it does not come in with the `kwargs` so we need to
         # set it here in order for the following logic to capture it
         if "source" not in kwargs and hasattr(self, "source"):
             kwargs["source"] = self.source  # type: ignore
 
-        if "source" in kwargs and "with_param" in kwargs:
-            arguments += self._get_deduped_params_from_source(parameter_names, artifact_names, kwargs["source"])
-        elif "source" in kwargs and "with_items" in kwargs:
-            arguments += self._get_deduped_params_from_items(parameter_names, kwargs["with_items"])
+        if hasattr(self, "directly_callable") and self.directly_callable:
+            function_kwargs = {k: v for k, v in kwargs.items() if k not in ["source", "name"]}
+            arguments = self._extract_arguments_directly_callable(args, function_kwargs)
```

Comment:
We're not necessarily dealing with a function here - you have examples of `directly_callable` `Containers`